### PR TITLE
MAM-2372-Layer-2-MVP

### PR DESCRIPTION
**Pull request by jtomchak**
Create the 'engagement follow feed' of those direct follows for users
- [x] This will need to be a reoccurring job
- [x] fetch mammoth accounts from acctrelay
- [x] for each account fetch following
- [x] get statuses for following and then → send them on to for_you_feed_worker
- [x] get statuses for _indirect_ following and then → send them on to for_you_feed_worker
- [x] Engagement numbers need to be set and then filter those statuses
- [x] It will run, sending it's output to the user's Personalized For You Feed set
- [x] This should be a subtask of personalfeed worker. It can be done in parallel, and we don't want each layer to be it's own cron job.

### CODE REVIEW

File: app/workers/scheduler/status_stat_update_scheduler.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@
 class Scheduler::StatusStatUpdateScheduler
   FOR_YOU_OWNER_ACCOUNT = ENV['FOR_YOU_OWNER_ACCOUNT'] || 'admin'
   LIST_TITLE = 'For You'
-  GO_BACK = 24 # number of hours back to fetch statuses
+  GO_BACK = 12 # number of hours back to fetch statuses
```
Comment (rileyhCode) [inline app/workers/scheduler/status_stat_update_scheduler.rb:9]: Was this due to the processing taking too long? Or just no need to go back 24 hours for some other reason?

File: app/workers/update_for_you_worker.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    @personal.statuses_for_direct_follows(@acct)
+             .filter_map { |s| engagment_threshold(s, user_setting[:your_follows], 'following') }
+             .each { |s| ForYouFeedWorker.perform_async(s['id'], @account.id, 'personal') } unless user_setting[:your_follows].zero?
+  end
+
+  # Indirect Follows
+  def push_indirect_following_status
+    user_setting = @user[:for_you_settings]
+    return if user_setting[:friends_of_friends].zero?
```
Comment (rileyhCode) [inline app/workers/update_for_you_worker.rb:79]: Not sure it matters, but… the code above uses the " unless ___" construct, and this code uses a "return if ___" construct. Maybe pick one style to use in both places. (?)

**Review by rileyhCode — APPROVED** [review]
Added some comments. Feel free to merge once you take a look.

File: app/workers/scheduler/status_stat_update_scheduler.rb (hunk 1 above)
Comment (jtomchak) [reply app/workers/scheduler/status_stat_update_scheduler.rb:9]: with the job running every 10 minutes it felt unnecessary to run the job 6 times an hour, every hour, and still go back a full day

File: app/workers/update_for_you_worker.rb (hunk 2 above)
Comment (jtomchak) [reply app/workers/update_for_you_worker.rb:79]: oh. I had meant to change that back so they both had an early return. good call

**Pull request merged**